src/test/java/org/mockitousage/matchers/CapturingArgumentsTest.java

```diff
@@ -298,7 +313,7 @@ public void captures_correctly_when_captor_used_multiple_times() throws Exceptio
 
         // then
         // this is only for backwards compatibility. It does not make sense in real to do so.
-        verify(mock).mixedVarargs(any(), argumentCaptor.capture(), argumentCaptor.capture(), argumentCaptor.capture());
+        verify(mock).mixedVarargs(any(), argumentCaptor.capture());
```

Comment:
@ChristianSchwarz Hi, I have a proof of concept. I added a `hashCode` and `equals` method to each matcher. When we use `capture` or `captureIf` we create a new instance of `CapturingMatcher` only of that matcher was never used. I will commit when it's ready. What do you think?